docs/src/use/core-concepts.md

```diff
@@ -23,6 +23,23 @@ ESLint contains hundreds of built-in rules that you can use. You can also create
 
 For more information, refer to [Rules](../rules/).
 
+### Rule Fixes
+
+Rules may optionally provide fixes for violations that they find. Fixes safely correct the violation without changing application logic.
```

Comment:
https://github.com/eslint/eslint/pull/17657#pullrequestreview-1682571543:

> I've started to use the term "violations" to describe lint errors -- we have too many other words floating around in the docs right now, so definitely room to improve.

Interesting! Are there any other discussion places that come to mind for the consolidation? I can file an issue if not. IMO it'd be valuable to have shared vocab used as a standard in the docs. Even a glossary long-term.

In case it's useful: "violation" is not what I personally would lean towards right now. It's a bit more intense and opinionated of a word than "report".

Reply:
Nothing official, no. Just something I've been thinking about. A glossary is definitely of interest to me.

Our problem is that we already have clearly defined meanings for several terms that we could use:

1. problem - Type of rule
2. suggestion - Actually both a type of rule and an unsafe fix (need to disambiguate this in the future)
3. error - Too ambiguous vs. runtime errors and also a severity level
4. warning - Severity level

We've actually used "report" historically to represent the collection of violations for a given run, so I'm not in favor of repurposing that.

Thinking in terms of rule violations make sense to me, which is why I started adopting "violation" as the term. Definitely open to other suggestions.

Reply:
Realized I never back-linked: https://github.com/eslint/eslint/issues/17987 tracks adding a glossary.